File: src/main/java/hilfling/backend/hilfing/service/GangService.java
package hilfling.backend.hilfing.service;

import hilfling.backend.hilfing.model.Gang;
import hilfling.backend.hilfing.repositories.GangRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GangService {
    @Autowired
    GangRepository gangRepository;

    public ResponseEntity<Gang> createGang(Gang gang) {
        try {
            Gang savedGang = gangRepository.save(gang);
            return ResponseEntity.ok().body(savedGang);
        } catch (Exception e) {
            return ResponseEntity.status(304).build();
        }
    }

    public ResponseEntity<Gang> getGang(Long id) {
        return gangRepository.findById(id)
                .map(gang -> ResponseEntity.ok().body(gang))
                .orElse(ResponseEntity.notFound().build());
    }

    public ResponseEntity<List<Gang>> getAllGangs() {
        List<Gang> gangs = gangRepository.findAll();
        return ResponseEntity.ok().body(gangs);
    }

    public ResponseEntity<Gang> updateGang(Gang gangDetails, Long id) {
        return gangRepository.findById(id)
                .map(gang -> {
                    gang.setName(gangDetails.getName());
                    Gang updatedGang = gangRepository.save(gang);
                    return ResponseEntity.ok().body(updatedGang);
                }).orElse(ResponseEntity.notFound().build());
    }

    public ResponseEntity<?> deleteGang(Long id) {
        return gangRepository.findById(id)
                .map(gang -> {
                    gangRepository.deleteById(id);
                    return ResponseEntity.ok().build();
                }).orElse(ResponseEntity.notFound().build());
    }
}

File: src/main/kotlin/no/fg/hilflingbackend/service/GangService.kt
package no.fg.hilflingbackend.service

import jakarta.persistence.EntityNotFoundException
import no.fg.hilflingbackend.dto.GangDto
import no.fg.hilflingbackend.dto.GangPatchRequestDto
import no.fg.hilflingbackend.dto.Page
import no.fg.hilflingbackend.repository.GangRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class GangService(
  val repository: GangRepository,
) {
  fun findById(id: UUID): GangDto = repository.findById(id) ?: throw EntityNotFoundException("Gang $id not found")

  fun findAll(
    page: Int,
    pageSize: Int,
  ): Page<GangDto> = repository.findAll(page, pageSize)

  fun create(dto: GangDto): Int = repository.create(dto)

  fun delete(id: UUID): Int = repository.delete(id)

  fun patch(dto: GangPatchRequestDto): GangDto = repository.patch(dto)
}
